wurst/dummy/InstantDummyCaster.wurst

```diff
@@ -7,111 +7,118 @@ import UnitObjEditing
-		return castPoint(owner, abilityId, lvl, orderId, targetPos, targetPos)
+		return castPoint(owner, abilityId, lvl, orderId, targetPos, targetPos.toVec3())
 
-	// string order wrappers
-	static function castImmediate(player owner, int abilityId, int lvl, string order, vec2 pos) returns boolean
+	// string orders / vec2 casterpos wrappers
+	static function castImmediate(player owner, int abilityId, int lvl, string order, vec3 pos) returns boolean
 		return castImmediate(owner, abilityId, lvl, OrderId(order), pos)
-
-	static function castTarget(player owner, int abilityId, int lvl, string order, widget target, vec2 casterPos) returns boolean
-		return castTarget(owner, abilityId, lvl, OrderId(order), target, casterPos)
+	static function castImmediate(player owner, int abilityId, int lvl, string order, vec2 pos) returns boolean
+		return castImmediate(owner, abilityId, lvl, OrderId(order), pos.toVec3())
+	static function castImmediate(player owner, int abilityId, int lvl, int orderId, vec2 pos) returns boolean
+		return castImmediate(owner, abilityId, lvl, orderId, pos.toVec3())
 
 	static function castTarget(player owner, int abilityId, int lvl, string order, widget target) returns boolean
 		return castTarget(owner, abilityId, lvl, OrderId(order), target)
-		
-	static function castPoint(player owner, int abilityId, int lvl, string order, vec2 targetPos, vec2 casterPos) returns boolean
-		return castPoint(owner, abilityId, lvl, OrderId(order), targetPos, casterPos)
-		
+	static function castTarget(player owner, int abilityId, int lvl, string order, widget target, vec3 casterPos) returns boolean
+		return castTarget(owner, abilityId, lvl, OrderId(order), target, casterPos)
+	static function castTarget(player owner, int abilityId, int lvl, string order, widget target, vec2 casterPos) returns boolean
+		return castTarget(owner, abilityId, lvl, OrderId(order), target, casterPos.toVec3())
+	static function castTarget(player owner, int abilityId, int lvl, int orderId, widget target, vec2 casterPos) returns boolean
+		return castTarget(owner, abilityId, lvl, orderId, target, casterPos.toVec3())
+
 	static function castPoint(player owner, int abilityId, int lvl, string order, vec2 targetPos) returns boolean
 		return castPoint(owner, abilityId, lvl, OrderId(order), targetPos)
-		
+	static function castPoint(player owner, int abilityId, int lvl, string order, vec2 targetPos, vec3 casterPos) returns boolean
+		return castPoint(owner, abilityId, lvl, OrderId(order), targetPos, casterPos)
+	static function castPoint(player owner, int abilityId, int lvl, string order, vec2 targetPos, vec2 casterPos) returns boolean
+		return castPoint(owner, abilityId, lvl, OrderId(order), targetPos, casterPos.toVec3())
+	static function castPoint(player owner, int abilityId, int lvl, int orderId, vec2 targetPos, vec2 casterPos) returns boolean
+		return castPoint(owner, abilityId, lvl, orderId, targetPos, casterPos.toVec3())
+
 constant DUMMY_CASTER_UNIT_ID = 'xcas'
```

Comment:
should probably use compiletime expression with id gen

Reply:
I guess I know why it wasn't done in the first place... ok, will do.